presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -1145,12 +1160,23 @@ public void dropMaterializedView(Session session, QualifiedObjectName viewName)
     }
 
     @Override
-    public MaterializedViewFreshness getMaterializedViewFreshness(Session session, TableHandle tableHandle)
+    public MaterializedViewFreshness getMaterializedViewFreshness(Session session, QualifiedObjectName viewName)
     {
-        CatalogName catalogName = tableHandle.getCatalogName();
-        CatalogMetadata catalogMetadata = getCatalogMetadata(session, catalogName);
-        ConnectorMetadata metadata = catalogMetadata.getMetadataFor(catalogName);
-        return metadata.getMaterializedViewFreshness(session.toConnectorSession(catalogName), tableHandle.getConnectorHandle());
+        if (viewName.getCatalogName().isEmpty() || viewName.getSchemaName().isEmpty() || viewName.getObjectName().isEmpty()) {
```

Comment:
How is it possible for all of them to be empty? I don't think it's valid for any of them to be empty.

Reply:
Yes, constructor for `QualifiedObjectName` requires the three components to be non-empty. I removed the check.